src/libsystemd/sd-device/sd-device.c

```diff
@@ -325,6 +325,10 @@ _public_ int sd_device_new_from_subsystem_sysname(sd_device **ret, const char *s
         if (access(syspath, F_OK) >= 0)
                 return sd_device_new_from_syspath(ret, syspath);
 
+        syspath = strjoina("/sys/firmware/", subsystem, "/", sysname);
```

Comment:
don't know much about device tree, but this generally looks ok. Main question is: is there any chance of conflict when looking up something that doesn't use "devicetree/base"?

e.g. I have a /sys/firmware/dmi path which has different contents to the /sys/class/dmi path. We match on the latter first so this patch would have no effect on current rules, but now I'm wondering if there's something that we want to match in /sys/firmware that is effectively shadowed by one of the earlier rules?